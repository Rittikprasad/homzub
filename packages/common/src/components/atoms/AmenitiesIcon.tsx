import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import Icon from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { ITypographyProps, Typography } from '@homzhub/common/src/components/atoms/Typography';

interface IProps {
  direction: 'row' | 'column';
  icon: string;
  iconSize?: number;
  iconColor?: string;
  label: string;
  labelStyles?: ITypographyProps;
  isLastIndex?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
}

const AmenitiesIcon = (props: IProps): React.ReactElement => {
  const {
    direction,
    icon,
    iconSize = 40,
    iconColor = theme.colors.darkTint3,
    label,
    isLastIndex = false,
    containerStyle,
    labelStyles,
  } = props;

  const renderText = (): React.ReactElement => {
    const labelStyle = direction === 'column' ? styles.columnLabel : styles.rowLabel;
    return (
      <Typography
        variant={PlatformUtils.isWeb() ? 'label' : labelStyles ? labelStyles.variant : 'text'}
        size={PlatformUtils.isWeb() ? 'regular' : labelStyles ? labelStyles.size : 'small'}
        fontWeight="regular"
        style={[styles.label, labelStyle]}
        minimumFontScale={0.5}
        adjustsFontSizeToFit
      >
        {label}
      </Typography>
    );
  };

  const renderColumn = (): React.ReactElement => {
    return (
      <>
        <View style={[styles.columnContainer, containerStyle]}>
          <Icon name={icon} size={iconSize} color={iconColor} />
          {renderText()}
        </View>
        {!isLastIndex && direction === 'column' && <Divider containerStyles={styles.divider} />}
      </>
    );
  };

  const renderRow = (): React.ReactElement => {
    return (
      <View style={[styles.rowContainer, containerStyle]}>
        <Icon name={icon} size={iconSize} color={iconColor} />
        {renderText()}
      </View>
    );
  };

  return direction === 'column' ? renderColumn() : renderRow();
};

export { AmenitiesIcon };

const styles = StyleSheet.create({
  columnContainer: {
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'center',
    marginHorizontal: 10,
  },
  rowContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 4,
  },
  label: {
    color: theme.colors.darkTint3,
  },
  columnLabel: {
    alignItems: 'center',
  },
  rowLabel: {
    marginStart: 6,
  },
  divider: {
    borderWidth: 1,
    borderColor: theme.colors.background,
    height: 25,
  },
});
