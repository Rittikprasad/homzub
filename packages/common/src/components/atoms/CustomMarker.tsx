import React from 'react';
import { View, StyleSheet } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';

interface IProps {
  selected: boolean;
}

const CustomMarker = (props: IProps): React.ReactElement => {
  const { selected } = props;

  if (!selected) {
    return <View style={styles.marker} />;
  }

  return (
    <View style={styles.selectedMarker}>
      <View style={styles.marker} />
    </View>
  );
};

const styles = StyleSheet.create({
  marker: {
    ...(theme.circleCSS(16) as object),
    backgroundColor: theme.colors.primaryColor,
  },
  selectedMarker: {
    ...(theme.circleCSS(40) as object),
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.markerOpacity,
  },
});

const memoizedComponent = React.memo(CustomMarker);
export { memoizedComponent as CustomMarker };
