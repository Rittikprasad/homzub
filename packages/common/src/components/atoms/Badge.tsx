import React from 'react';
import { StyleProp, StyleSheet, TextStyle, View, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { FontWeightType, Label } from '@homzhub/common/src/components/atoms/Text';

interface IProps {
  title: string;
  badgeColor: string;
  badgeStyle?: StyleProp<ViewStyle>;
  textType?: FontWeightType;
  titleStyle?: StyleProp<TextStyle>;
}

export const Badge = (props: IProps): React.ReactElement => {
  const { badgeColor, badgeStyle, title, textType = 'semiBold', titleStyle = {} } = props;
  return (
    <View style={[styles.badge, badgeStyle, { backgroundColor: badgeColor }]}>
      <Label type="regular" textType={textType} style={[styles.title, titleStyle]}>
        {title}
      </Label>
    </View>
  );
};

const styles = StyleSheet.create({
  badge: {
    borderRadius: 2,
    paddingHorizontal: 6,
  },
  title: {
    color: theme.colors.white,
    paddingBottom: 0,
    alignSelf: 'center',
  },
});
