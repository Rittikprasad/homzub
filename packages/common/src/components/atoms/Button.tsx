import React from 'react';
import {
  GestureResponderEvent,
  ImageStyle,
  StyleProp,
  StyleSheet,
  TextProps,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon from '@homzhub/common/src/assets/icon';
import { FontWeightType, Label, Text, TextFieldType, TextSizeType } from '@homzhub/common/src/components/atoms/Text';

export type ButtonType = 'primary' | 'secondary' | 'secondaryOutline' | 'text';

export interface IButtonProps {
  type: ButtonType;
  node?: React.ReactNode;
  children?: React.ReactNode;
  title?: string;
  onPress?: (event?: GestureResponderEvent) => void;
  disabled?: boolean;
  activeOpacity?: number;
  titleStyle?: StyleProp<TextStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  textType?: TextFieldType;
  textSize?: TextSizeType;
  fontType?: FontWeightType;
  icon?: string;
  iconSize?: number;
  iconColor?: string;
  iconStyle?: StyleProp<ImageStyle>;
  testID?: string;
  maxLength?: number;
  textStyle?: TextStyle;
  numOfLines?: number;
  ellipsizeMode?: TextProps['ellipsizeMode'];
}

export class Button extends React.PureComponent<IButtonProps> {
  public render = (): React.ReactElement => {
    const {
      onPress,
      disabled = false,
      activeOpacity = 0.5,
      title,
      icon,
      iconSize,
      iconColor,
      iconStyle,
      testID,
      node,
      children,
    } = this.props;
    return (
      <TouchableOpacity
        activeOpacity={activeOpacity}
        onPress={onPress}
        disabled={disabled}
        style={this.getContainerStyle()}
        testID={testID}
      >
        {title && this.getTextField()}
        {!!icon && <Icon name={icon} size={iconSize} color={iconColor} style={iconStyle} />}
        {node && node}
        {children && children}
      </TouchableOpacity>
    );
  };

  private getTextField = (): React.ReactElement => {
    const {
      textType,
      title,
      textSize = 'small',
      fontType = 'semiBold',
      maxLength,
      textStyle,
      numOfLines,
      ellipsizeMode = 'tail',
    } = this.props;
    let TextField = Text;

    if (textType === 'label') {
      TextField = Label;
    }

    return (
      <TextField
        type={textSize}
        textType={fontType}
        style={[this.getTextStyle(), textStyle && textStyle]}
        maxLength={maxLength}
        ellipsizeMode={ellipsizeMode}
        numberOfLines={numOfLines}
      >
        {title}
      </TextField>
    );
  };

  private getContainerStyle = (): ViewStyle => {
    const { disabled, type, containerStyle = {} } = this.props;
    let themedStyle: StyleProp<ViewStyle> = styles.primary;
    if (disabled) {
      themedStyle = styles.disabled;
    }

    if (type === 'secondary' && !disabled) {
      themedStyle = styles.secondary;
    }

    if (type === 'secondaryOutline' && !disabled) {
      themedStyle = styles.secondaryOutline;
    }

    if (type === 'text') {
      themedStyle = styles.textButton;
    }

    return StyleSheet.flatten([styles.container, themedStyle, containerStyle]);
  };

  private getTextStyle = (): TextStyle => {
    const { type, disabled = false, titleStyle = {} } = this.props;
    let themedStyle = {};
    if (type === 'secondary' && !disabled) {
      themedStyle = { color: theme.colors.active };
    }
    if (type === 'text' && !disabled) {
      themedStyle = { color: theme.colors.darkTint2 };
    }
    if (type === 'text' && disabled) {
      themedStyle = { color: theme.colors.darkTint8 };
    }

    return StyleSheet.flatten([styles.textStyle, themedStyle, titleStyle]);
  };
}

const styles = StyleSheet.create({
  container: {
    ...theme.globalStyles.center,
    borderRadius: 4,
  },
  textStyle: {
    textAlign: 'center',
    color: theme.colors.white,
    marginVertical: 12,
    marginHorizontal: 18,
  },
  disabled: {
    backgroundColor: theme.colors.disabled,
  },
  primary: {
    backgroundColor: theme.colors.primaryColor,
  },
  secondary: {
    borderWidth: 1,
    borderColor: theme.colors.active,
    backgroundColor: theme.colors.white,
  },
  secondaryOutline: {
    borderWidth: 0.5,
    borderColor: theme.colors.white,
  },
  textButton: {
    borderWidth: 0,
  },
});
