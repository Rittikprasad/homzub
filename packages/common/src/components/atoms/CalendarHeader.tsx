import React from 'react';
import { StyleSheet, View } from 'react-native';
import moment from 'moment';
import { FunctionUtils } from '@homzhub/common/src/utils/FunctionUtils';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { Text } from '@homzhub/common/src/components/atoms/Text';

interface IProps {
  isAllowPastDate?: boolean;
  headerTitle?: string;
  headerYear?: string;
  yearTitle?: string;
  isMonthView?: boolean;
  isYearView?: boolean;
  isCurrentMonth?: boolean;
  maxDate?: string;
  minDate?: string;
  month?: number;
  year?: string;
  onBackPress?: () => void;
  onNextPress?: () => void;
  onMonthPress?: () => void;
  onYearPress?: () => void;
}

const CalendarHeader = (props: IProps): React.ReactElement => {
  const {
    isAllowPastDate,
    maxDate,
    minDate,
    onBackPress,
    onNextPress,
    onMonthPress,
    onYearPress,
    isCurrentMonth,
    headerTitle,
    headerYear,
    isMonthView = false,
    isYearView = false,
    yearTitle,
    month,
    year,
  } = props;

  // DISABLE VALIDATION
  const isNextDisable =
    maxDate &&
    (moment(maxDate).month() === moment().month() ||
      (!!month && moment(maxDate).month() === month && !!year && moment(maxDate).year() === Number(year)));
  const isPreviousDisable =
    (minDate ? moment(minDate).year() === moment().year() : !isAllowPastDate) && isCurrentMonth && !isYearView;
  // DISABLE VALIDATION

  return (
    <>
      <View style={styles.headerContainer}>
        <Icon
          name={icons.leftArrow}
          onPress={!isPreviousDisable ? onBackPress : FunctionUtils.noop}
          size={22}
          color={isPreviousDisable ? theme.colors.disabled : theme.colors.primaryColor}
        />
        {!isYearView && !isMonthView && (
          <Text type="small" textType="semiBold" onPress={onMonthPress} style={styles.headerTitle}>
            {headerTitle}
          </Text>
        )}
        {isMonthView && (
          <Text type="small" textType="semiBold" onPress={onYearPress} style={styles.headerTitle}>
            {headerYear}
          </Text>
        )}
        {isYearView && (
          <Text type="small" textType="semiBold" style={styles.headerTitle}>
            {yearTitle}
          </Text>
        )}
        <Icon
          name={icons.rightArrow}
          size={22}
          color={isNextDisable ? theme.colors.disabled : theme.colors.primaryColor}
          onPress={!isNextDisable ? onNextPress : FunctionUtils.noop}
        />
      </View>
      <Divider />
    </>
  );
};

export default CalendarHeader;

const styles = StyleSheet.create({
  headerContainer: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    paddingVertical: 20,
    paddingHorizontal: 24,
  },
  headerTitle: {
    color: theme.colors.primaryColor,
  },
});
