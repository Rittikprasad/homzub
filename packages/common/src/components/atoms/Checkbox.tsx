import React from 'react';
import { StyleProp, StyleSheet, TextStyle, TouchableOpacity, ViewStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Label, TextSizeType } from '@homzhub/common/src/components/atoms/Text';

interface ICheckboxOptions {
  selected: boolean;
  label?: string;
  onToggle: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  iconSize?: number;
  iconStyle?: object;
  iconSelectedStyle?: object;
  labelType?: TextSizeType;
  testID?: string;
  subLabel?: string;
  onPressLink?: () => void;
}

const RNCheckbox = (props: ICheckboxOptions): React.ReactElement => {
  const {
    label,
    selected,
    containerStyle = {},
    labelStyle = {},
    iconSize = 22,
    iconStyle = {},
    iconSelectedStyle = {},
    labelType = 'large',
    onToggle,
    testID,
    subLabel,
    onPressLink,
  } = props;
  return (
    <TouchableOpacity style={[styles.container, containerStyle]} onPress={onToggle} testID={testID}>
      <Icon
        name={selected ? icons.checkboxOn : icons.checkboxOff}
        size={iconSize}
        color={selected ? theme.colors.primaryColor : theme.colors.disabled}
        style={selected ? iconSelectedStyle : iconStyle}
      />
      {!!label && (
        <Label type={labelType} textType="regular" style={[styles.label, labelStyle]}>
          {label}
          {!!subLabel && (
            <Label type="regular" style={styles.subLabel} onPress={onPressLink}>
              {subLabel}
            </Label>
          )}
        </Label>
      )}
    </TouchableOpacity>
  );
};

export { RNCheckbox };

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  label: {
    marginStart: 12,
    marginEnd: 4,
    flex: 1,
  },
  subLabel: {
    color: theme.colors.primaryColor,
  },
});
