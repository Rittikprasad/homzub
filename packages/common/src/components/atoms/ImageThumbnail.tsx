import React from 'react';
import { ImageBackground, ImageStyle, StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { Typography } from '@homzhub/common/src/components/atoms/Typography';

export interface IOwnProps {
  imageUrl: null | string;
  onIconPress?: () => void;
  dataLength?: number;
  iconSize?: number;
  iconColor?: string;
  containerStyle?: StyleProp<ViewStyle>;
  imageContainerStyle?: StyleProp<ImageStyle>;
  imageWrapperStyle?: StyleProp<ImageStyle>;
  isIconVisible?: boolean;
  isCoverPhotoContainer?: boolean;
  coverPhotoTitle?: string;
  isLastThumbnail?: boolean;
  onPressLastThumbnail?: () => void;
  isFavorite?: boolean;
  markFavorite?: () => void;
  galleryView?: boolean;
  onPressImage?: () => void;
}

type IProps = WithTranslation & IOwnProps;

class ImageThumbnail extends React.PureComponent<IProps> {
  public render(): React.ReactElement {
    const {
      imageUrl,
      iconSize,
      iconColor,
      imageContainerStyle,
      imageWrapperStyle,
      containerStyle,
      isIconVisible = true,
      isCoverPhotoContainer = false,
      isLastThumbnail = false,
      onPressLastThumbnail,
      dataLength,
      isFavorite = false,
      coverPhotoTitle = 'Cover Photo',
      markFavorite,
      t,
      galleryView,
      onPressImage,
    } = this.props;

    const { handleIconPress } = this;
    const Content = React.memo(
      (): React.ReactElement => (
        <>
          {!!imageUrl && (
            <ImageBackground
              resizeMode={PlatformUtils.isWeb() && galleryView ? 'contain' : 'cover'}
              source={{ uri: imageUrl }}
              imageStyle={[styles.imageStyle, imageContainerStyle]}
              style={[styles.imageWrapper, imageWrapperStyle]}
            >
              {isIconVisible && (
                <TouchableOpacity
                  style={
                    PlatformUtils.isWeb() && isCoverPhotoContainer ? styles.iconContainerWeb : styles.iconContainer
                  }
                  onPress={handleIconPress}
                >
                  <Icon name={icons.close} size={iconSize || 22} color={iconColor || theme.colors.white} />
                  {PlatformUtils.isWeb() && isCoverPhotoContainer && (
                    <Typography variant="label" size="large" style={styles.removeTxt}>
                      {t('remove')}
                    </Typography>
                  )}
                </TouchableOpacity>
              )}
              {isCoverPhotoContainer && (
                <TouchableOpacity style={styles.touchableContainer} onPress={markFavorite}>
                  <View style={styles.coverPhotoContainer}>
                    <Text type="small" textType="regular" style={styles.coverPhoto}>
                      {coverPhotoTitle}
                    </Text>
                    <Icon
                      name={isFavorite ? icons.starFilled : icons.starUnfilled}
                      size={20}
                      color={theme.colors.white}
                      style={styles.starIcon}
                    />
                  </View>
                </TouchableOpacity>
              )}
              {isLastThumbnail && (
                <View style={styles.lastThumbnailContainer}>
                  <TouchableOpacity onPress={onPressLastThumbnail}>
                    <Text type="regular" textType="semiBold" style={styles.numberOfImages}>
                      + {dataLength}
                    </Text>
                  </TouchableOpacity>
                </View>
              )}
            </ImageBackground>
          )}
        </>
      )
    );

    if (onPressImage) {
      return (
        <TouchableOpacity style={[styles.container, containerStyle]} onPress={onPressImage}>
          <Content />
        </TouchableOpacity>
      );
    }
    return (
      <View style={[styles.container, containerStyle]}>
        <Content />
      </View>
    );
  }

  public handleIconPress = (): void => {
    const { onIconPress } = this.props;
    if (onIconPress) {
      onIconPress();
    }
  };
}

export default withTranslation()(ImageThumbnail);

const styles = StyleSheet.create({
  container: {
    marginTop: 10,
  },
  imageStyle: {
    borderRadius: 4,
  },
  imageWrapper: {
    height: 200,
  },
  iconContainer: {
    width: 30,
    height: 30,
    borderRadius: 30,
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    top: 10,
    right: 10,
    bottom: 0,
    backgroundColor: theme.colors.crossIconContainer,
  },
  iconContainerWeb: {
    width: 'fit-content',
    height: 30,
    flexDirection: 'row',
    borderRadius: 2,
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    paddingHorizontal: 8,
    top: 10,
    right: 10,
    bottom: 0,
    backgroundColor: theme.colors.crossIconContainerWeb,
  },
  removeTxt: {
    color: 'white',
  },
  touchableContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
  },
  coverPhotoContainer: {
    flexDirection: 'row',
    backgroundColor: theme.colors.imageThumbnailBackground,
    opacity: 0.8,
    height: 35,
    alignItems: 'center',
    paddingLeft: 10,
    borderRadius: 4,
  },
  coverPhoto: { color: theme.colors.white, opacity: 1, flex: 7 },
  lastThumbnailContainer: {
    flexDirection: 'row',
    backgroundColor: theme.colors.darkTint1,
    opacity: 0.75,
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 4,
  },
  numberOfImages: {
    color: theme.colors.white,
    opacity: 1,
    justifyContent: 'center',
    padding: 30,
  },
  starIcon: {
    flex: 1,
  },
});
