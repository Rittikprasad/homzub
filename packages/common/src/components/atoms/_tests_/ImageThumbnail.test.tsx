import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ImageThumbnail, IOwnProps } from '@homzhub/common/src/components/atoms/ImageThumbnail';

describe.skip('Image Thumbnail', () => {
  let props: IOwnProps;

  it('should match snapshot for isCoverPhotoContainer', () => {
    props = {
      imageUrl: 'image',
      isCoverPhotoContainer: true,
    };
    const wrapper: ShallowWrapper = shallow(<ImageThumbnail {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for isCoverPhotoContainer when isFavourite is true ', () => {
    props = {
      imageUrl: 'image',
      isCoverPhotoContainer: true,
      isFavorite: true,
    };
    const wrapper: ShallowWrapper = shallow(<ImageThumbnail {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for isLastThumbnail ', () => {
    props = {
      imageUrl: 'image',
      isLastThumbnail: true,
    };
    const wrapper: ShallowWrapper = shallow(<ImageThumbnail {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for handleIconPress when onIconPress prop is there ', () => {
    props = {
      imageUrl: 'image',
      onIconPress: jest.fn(),
    };
    const wrapper: ShallowWrapper = shallow(<ImageThumbnail {...props} />);
    // @ts-ignore
    wrapper.instance().handleIconPress();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot for handleIconPress when onIconPress prop is not there', () => {
    props = {
      imageUrl: 'image',
    };
    const wrapper: ShallowWrapper = shallow(<ImageThumbnail {...props} />);
    // @ts-ignore
    wrapper.instance().handleIconPress();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
