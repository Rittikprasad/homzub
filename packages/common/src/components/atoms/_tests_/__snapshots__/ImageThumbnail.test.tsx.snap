// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Thumbnail should match snapshot for handleIconPress when onIconPress prop is not there 1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    imageStyle={
      Array [
        Object {
          "borderRadius": 4,
        },
        undefined,
      ]
    }
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Array [
        Object {
          "height": 200,
        },
        undefined,
      ]
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 30,
          "bottom": 0,
          "height": 30,
          "justifyContent": "center",
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": 30,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#FFFFFF"
        name="close"
        size={22}
      />
    </ForwardRef>
  </ImageBackground>
</View>
`;

exports[`Image Thumbnail should match snapshot for handleIconPress when onIconPress prop is there  1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    imageStyle={
      Array [
        Object {
          "borderRadius": 4,
        },
        undefined,
      ]
    }
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Array [
        Object {
          "height": 200,
        },
        undefined,
      ]
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 30,
          "bottom": 0,
          "height": 30,
          "justifyContent": "center",
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": 30,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#FFFFFF"
        name="close"
        size={22}
      />
    </ForwardRef>
  </ImageBackground>
</View>
`;

exports[`Image Thumbnail should match snapshot for isCoverPhotoContainer 1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    imageStyle={
      Array [
        Object {
          "borderRadius": 4,
        },
        undefined,
      ]
    }
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Array [
        Object {
          "height": 200,
        },
        undefined,
      ]
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 30,
          "bottom": 0,
          "height": 30,
          "justifyContent": "center",
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": 30,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#FFFFFF"
        name="close"
        size={22}
      />
    </ForwardRef>
    <ForwardRef
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 27, 54, 0.6)",
            "borderRadius": 4,
            "flexDirection": "row",
            "height": 35,
            "opacity": 0.8,
            "paddingLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "flex": 7,
              "opacity": 1,
            }
          }
          textType="regular"
          type="small"
        >
          Cover Photo
        </Text>
        <Icon
          allowFontScaling={false}
          color="#FFFFFF"
          name="unfilled-star"
          size={20}
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </ForwardRef>
  </ImageBackground>
</View>
`;

exports[`Image Thumbnail should match snapshot for isCoverPhotoContainer when isFavourite is true  1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    imageStyle={
      Array [
        Object {
          "borderRadius": 4,
        },
        undefined,
      ]
    }
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Array [
        Object {
          "height": 200,
        },
        undefined,
      ]
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 30,
          "bottom": 0,
          "height": 30,
          "justifyContent": "center",
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": 30,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#FFFFFF"
        name="close"
        size={22}
      />
    </ForwardRef>
    <ForwardRef
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 27, 54, 0.6)",
            "borderRadius": 4,
            "flexDirection": "row",
            "height": 35,
            "opacity": 0.8,
            "paddingLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "flex": 7,
              "opacity": 1,
            }
          }
          textType="regular"
          type="small"
        >
          Cover Photo
        </Text>
        <Icon
          allowFontScaling={false}
          color="#FFFFFF"
          name="filled-star"
          size={20}
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </ForwardRef>
  </ImageBackground>
</View>
`;

exports[`Image Thumbnail should match snapshot for isLastThumbnail  1`] = `
<View
  style={
    Array [
      Object {
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <ImageBackground
    imageStyle={
      Array [
        Object {
          "borderRadius": 4,
        },
        undefined,
      ]
    }
    source={
      Object {
        "uri": "image",
      }
    }
    style={
      Array [
        Object {
          "height": 200,
        },
        undefined,
      ]
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 30,
          "bottom": 0,
          "height": 30,
          "justifyContent": "center",
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": 30,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#FFFFFF"
        name="close"
        size={22}
      />
    </ForwardRef>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#001B36",
          "borderRadius": 4,
          "bottom": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "opacity": 0.75,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ForwardRef>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "justifyContent": "center",
              "opacity": 1,
              "padding": 30,
            }
          }
          textType="semiBold"
          type="regular"
        >
          + 
        </Text>
      </ForwardRef>
    </View>
  </ImageBackground>
</View>
`;
